examples/plot_fdboxplot.py

```diff
@@ -0,0 +1,124 @@
+"""
+Functional Boxplot
+====================
+
+Shows the use of the functional Boxplot applied to the Canadian Weather dataset.
+"""
+
+# Author: Amanda Hernando Bernabé
+# License: MIT
+
+# sphinx_gallery_thumbnail_number = 4
+
+
+from fda import datasets
+from fda.grid import FDataGrid
+from fda.boxplot import fdboxplot
+from fda.depth_measures import band_depth, fraiman_muniz_depth
+import matplotlib.pyplot as plt
+import numpy as np
+import matplotlib.patches as mpatches
+from cycler import cycler
+import matplotlib
+
+##################################################################################
+# First, the Canadian Weather dataset is downloaded from the package 'fda' in CRAN.
+# It contains a FDataGrid with daily temperatures and precipitations, that is, it
+# has a 2-dimensional image. We are interested only in the daily average temperatures,
+# so another FDataGrid is constructed with the desired values.
+
+dataset = datasets.fetch_weather()
+fd = dataset["data"]
+fd_temperatures = FDataGrid(data_matrix=fd.data_matrix[:, :, 0], sample_points=fd.sample_points,
+                            dataset_label=fd.dataset_label, axes_labels=fd.axes_labels[0:2])
+############################################################################################
+# The data is plotted to show the curves we are working with. They are divided according to the
+# target. In this case, it includes the different climates to which the weather stations belong to.
+
+climates = dataset["target_names"]
+n_climates = len(climates)
+
+#Assigning the color to each of the samples according to the target.
+colormap = plt.cm.get_cmap('seismic')
+colors_by_climate = colormap(np.asarray(dataset["target"]) / (n_climates- 1))
+climate_colors = colormap(np.arange(n_climates) / (n_climates- 1))
+
+#Building the legend
+patches = []
+for i in range(n_climates):
+    patches.append(mpatches.Patch(color=climate_colors[i], label=climates[i]))
+
+#Plotting the curves and the legend with the desired colors.
+matplotlib.rcParams['axes.prop_cycle'] = cycler(color=colors_by_climate)
```

Comment:
Since the plot method of FDataGrid only calls once to matplotlib.pyplot.plot for all the curves in the FDataGrid, and this last method does not accept an array specifying the color of each sample, the only way to customize the colors is in this way.

Reply:
Then we should make the plot method more customizable.

Reply:
If a color parameter is added in the plot method, then the *kwargs should be checked in order not to contain another color parameter because it would crash if not. Or, it should be specified in the documentation and not check it?

Reply:
I am not sure of what you are trying to say.

Reply:
The implementation of the plot method for FDataGrid:
```
    def plot(self, fig=None, ax=None, nrows=None, ncols=None, **kwargs):
    
        fig, ax = self.generic_plotting_checks(fig, ax, nrows, ncols)

        if self.ndim_domain == 1:
            for i in range(self.ndim_image):
                ax[i].plot(self.sample_points[0], self.data_matrix[:, :, i].T, **kwargs)
        else:
            X = self.sample_points[0]
            Y = self.sample_points[1]
            X, Y = numpy.meshgrid(X, Y)
            for i in range(self.ndim_image):
                for j in range(self.nsamples):
                    ax[i].plot_surface(X, Y, numpy.squeeze(self.data_matrix[j, :, :, i]).T, **kwargs)

        self.set_labels(fig, ax)

        return fig, ax
```
If we want to accept a color parameter that includes a color for all the curves, the resulting method would be:
```
    def plot(self, fig=None, ax=None, nrows=None, ncols=None, color = None, **kwargs):
    
        fig, ax = self.generic_plotting_checks(fig, ax, nrows, ncols)
      
        if color is not None:
            check_kwargs

        if self.ndim_domain == 1:
            for i in range(self.ndim_image):
                for j in range(self.nsamples):
                    ax[i].plot(self.sample_points[0], self.data_matrix[j, :, i].T, color = color[j], **kwargs)
        else:
            X = self.sample_points[0]
            Y = self.sample_points[1]
            X, Y = numpy.meshgrid(X, Y)
            for i in range(self.ndim_image):
                for j in range(self.nsamples):
                    ax[i].plot_surface(X, Y, numpy.squeeze(self.data_matrix[j, :, :, i]).T, **kwargs)

        self.set_labels(fig, ax)

        return fig, ax
```

Reply:
The implementation of the plot method for FDataGrid:
```
    def plot(self, fig=None, ax=None, nrows=None, ncols=None, **kwargs):
    
        fig, ax = self.generic_plotting_checks(fig, ax, nrows, ncols)

        if self.ndim_domain == 1:
            for i in range(self.ndim_image):
                ax[i].plot(self.sample_points[0], self.data_matrix[:, :, i].T, **kwargs)
        else:
            X = self.sample_points[0]
            Y = self.sample_points[1]
            X, Y = numpy.meshgrid(X, Y)
            for i in range(self.ndim_image):
                for j in range(self.nsamples):
                    ax[i].plot_surface(X, Y, numpy.squeeze(self.data_matrix[j, :, :, i]).T, **kwargs)

        self.set_labels(fig, ax)

        return fig, ax
```
If we want to accept a color parameter that includes a color for all the curves, the resulting method would be:
```
    def plot(self, fig=None, ax=None, nrows=None, ncols=None, color = None, **kwargs):
    
        fig, ax = self.generic_plotting_checks(fig, ax, nrows, ncols)
      
        if color is not None:
            check_kwargs

        if self.ndim_domain == 1:
            for i in range(self.ndim_image):
                for j in range(self.nsamples):
                    ax[i].plot(self.sample_points[0], self.data_matrix[j, :, i].T, color = color[j], **kwargs)
        else:
            X = self.sample_points[0]
            Y = self.sample_points[1]
            X, Y = numpy.meshgrid(X, Y)
            for i in range(self.ndim_image):
                for j in range(self.nsamples):
                    ax[i].plot_surface(X, Y, numpy.squeeze(self.data_matrix[j, :, :, i]).T, **kwargs)

        self.set_labels(fig, ax)

        return fig, ax
```